Algorithm.CSharp/CustomUniverseWithBenchmarkRegressionAlgorithm.cs

```diff
@@ -0,0 +1,116 @@
+/*
+ * QUANTCONNECT.COM - Democratizing Finance, Empowering Individuals.
+ * Lean Algorithmic Trading Engine v2.0. Copyright 2014 QuantConnect Corporation.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+*/
+
+using System;
+using System.Collections.Generic;
+using QuantConnect.Data;
+using QuantConnect.Interfaces;
+
+namespace QuantConnect.Algorithm.CSharp
+{
+    /// <summary>
+    /// Regression algorithm with a custom universe and benchmark, both using the same security.
+    /// </summary>
+    public class CustomUniverseWithBenchmarkRegressionAlgorithm : QCAlgorithm, IRegressionAlgorithmDefinition
+    {
+        private readonly Symbol _spy = QuantConnect.Symbol.Create("SPY", SecurityType.Equity, Market.USA);
+        private const int ExpectedLeverage = 2;
+        private const int ExpectedDataPoints = 4;
+        private int _actualDatapoints;
+
+        /// <summary>
+        /// Initialise the data and resolution required, as well as the cash and start-end dates for your algorithm. All algorithms must initialized.
+        /// </summary>
+        public override void Initialize()
+        {
+            UniverseSettings.Resolution = Resolution.Daily;
+
+            SetStartDate(2014, 4, 2);  //Set Start Date
+            SetEndDate(2014, 4, 6);    //Set End Date
+            SetCash(100000);           //Set End Date
+
+            AddUniverse("my-universe", x => new List<string> { "SPY" });
```

Comment:
Can we make this have return empty for the first couple of days and then add it -- I want to confirm that there isn't a race condition between this universe and the benchmark universe in creating the security object which determines which one makes it into the algorithm's security dictionary.